test/AutoDiff/downstream/always_emit_into_client_multi_file.swift

```diff
@@ -0,0 +1,21 @@
+// TODO(TF-1103): Fix this test so that there is not a linker error. Then, move this test to
+// cross_module_derivative_attr_e2e.swift.
```

Comment:
I believe that this would belong in the validation tests directory

Reply:
What tests should go there? We haven't put any autodiff tests there -- we've put them all under `tests/`, even ones that build modules like this one.